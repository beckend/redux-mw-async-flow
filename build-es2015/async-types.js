"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultTypes = {
    ABORTED: '_ABORTED',
    END: '_END',
    FULFILLED: '_FULFILLED',
    PENDING: '_PENDING',
    REJECTED: '_REJECTED',
    REQUEST: '_REQUEST',
};
exports.getAsyncTypeConstants = ({ types } = {}) => (Object.assign({}, exports.defaultTypes, types));
exports.generateAsyncAction = ({ actionName, types = {} }) => {
    const { ABORTED, END, FULFILLED, PENDING, REJECTED, REQUEST, } = exports.getAsyncTypeConstants({ types });
    const ABORTED_NAME = `${actionName}${ABORTED}`;
    const END_NAME = `${actionName}${END}`;
    const FULFILLED_NAME = `${actionName}${FULFILLED}`;
    const PENDING_NAME = `${actionName}${PENDING}`;
    const REJECTEDT_NAME = `${actionName}${REJECTED}`;
    const REQUEST_NAME = `${actionName}${REQUEST}`;
    return {
        [ABORTED_NAME]: ABORTED_NAME,
        [END_NAME]: END_NAME,
        [FULFILLED_NAME]: FULFILLED_NAME,
        [PENDING_NAME]: PENDING_NAME,
        [REJECTEDT_NAME]: REJECTEDT_NAME,
        [REQUEST_NAME]: REQUEST_NAME,
    };
};
exports.replaceSuffix = (targetStr, suffixToReplace, replacementString) => `${targetStr.substring(0, targetStr.length - suffixToReplace.length)}${replacementString}`;
//# sourceMappingURL=async-types.js.map