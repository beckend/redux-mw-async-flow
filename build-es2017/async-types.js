"use strict";
const tslib_1 = require("tslib");
exports.defaultTypes = {
    REQUEST: 'REQUEST',
    PENDING: 'PENDING',
    FULFILLED: 'FULFILLED',
    REJECTED: 'REJECTED',
    ABORTED: 'ABORTED',
};
exports.getAsyncTypeConstants = ({ types } = {}) => {
    const { REQUEST, PENDING, FULFILLED, REJECTED, ABORTED, } = tslib_1.__assign({}, exports.defaultTypes, types);
    return {
        _REQUEST: `_${REQUEST}`,
        _PENDING: `_${PENDING}`,
        _FULFILLED: `_${FULFILLED}`,
        _REJECTED: `_${REJECTED}`,
        _ABORTED: `_${ABORTED}`,
    };
};
//# sourceMappingURL=async-types.js.map