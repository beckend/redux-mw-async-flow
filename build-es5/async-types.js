"use strict";
var tslib_1 = require("tslib");
exports.defaultTypes = {
    REQUEST: 'REQUEST',
    PENDING: 'PENDING',
    FULFILLED: 'FULFILLED',
    REJECTED: 'REJECTED',
    ABORTED: 'ABORTED',
};
exports.getAsyncTypeConstants = function (_a) {
    var types = (_a === void 0 ? {} : _a).types;
    var _b = tslib_1.__assign({}, exports.defaultTypes, types), REQUEST = _b.REQUEST, PENDING = _b.PENDING, FULFILLED = _b.FULFILLED, REJECTED = _b.REJECTED, ABORTED = _b.ABORTED;
    return {
        _REQUEST: "_" + REQUEST,
        _PENDING: "_" + PENDING,
        _FULFILLED: "_" + FULFILLED,
        _REJECTED: "_" + REJECTED,
        _ABORTED: "_" + ABORTED,
    };
};
//# sourceMappingURL=async-types.js.map