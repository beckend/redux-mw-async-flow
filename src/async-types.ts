export interface IDefaultTypes {
  readonly REQUEST: string;
  readonly PENDING: string;
  readonly FULFILLED: string;
  readonly REJECTED: string;
  readonly ABORTED: string;
}
export const defaultTypes: IDefaultTypes = {
  REQUEST: 'REQUEST',
  PENDING: 'PENDING',
  FULFILLED: 'FULFILLED',
  REJECTED: 'REJECTED',
  ABORTED: 'ABORTED',
};

export type TDefaultTypesOptional = {
  readonly[P in keyof IDefaultTypes]?: IDefaultTypes[P];
};
export interface IGetAsyncTypeConstantsParams {
  types?: TDefaultTypesOptional;
}
export const getAsyncTypeConstants = ({ types }: IGetAsyncTypeConstantsParams = {}) => {
  const {
    REQUEST,
    PENDING,
    FULFILLED,
    REJECTED,
    ABORTED,
  }: IDefaultTypes = {
      ...defaultTypes,
      ...types,
    };
  return {
    _REQUEST: `_${REQUEST}`,
    _PENDING: `_${PENDING}`,
    _FULFILLED: `_${FULFILLED}`,
    _REJECTED: `_${REJECTED}`,
    _ABORTED: `_${ABORTED}`,
  };
};
