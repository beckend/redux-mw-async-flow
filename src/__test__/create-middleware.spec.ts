/* tslint:disable: chai-vague-errors */
/* tslint:disable: max-line-length */
/* tslint:disable: no-increment-decrement */
// import { createStore, applyMiddleware } from 'redux';
import configureStore, { IStore } from 'redux-mock-store';
import {
  createAsyncFlowMiddleware,
  defaultOpts,
  TCreateAsyncFlowMiddlewareOpts,
  IAsyncFlowActionMeta,
} from '../create-middleware';
import { getAsyncTypeConstants } from '../async-types';
import { Action, ActionMeta } from 'redux-actions';

const {
  _REQUEST,
  // _PENDING,
  _FULFILLED,
  _REJECTED,
  _ABORTED,
} = getAsyncTypeConstants();

interface ICreateMockStoreOpts<TAction> {
  initialState?: any;
  asyncFlowOpts?: TCreateAsyncFlowMiddlewareOpts<TAction>;
}
const createMockStore = <TStoreState, TAction extends Action<any>>({ initialState, asyncFlowOpts }: ICreateMockStoreOpts<TAction> = {
  initialState: {},
}) => {
  return configureStore<any>([
    createAsyncFlowMiddleware<TStoreState, TAction>(asyncFlowOpts),
  ])(initialState);
};

describe('Middleware', () => {
  let mockStore: IStore<any>;
  beforeEach(() => {
    mockStore = createMockStore();
  });

  it('can be disabled, the action simply passes through', () => {
    const action: ActionMeta<{}, any> = {
      type: `ACTION${_REQUEST}`,
      meta: {
        [defaultOpts.metaKey]: {
          enable: false,
        } as IAsyncFlowActionMeta<any>,
      },
    };
    mockStore.dispatch(action);
    const actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(1);
    expect(actions[0])
      .toEqual(action);
  });

  it('actions that does not match suffix will pass through', () => {
    const action: Action<{}> = {
      type: 'ACTION',
    };
    mockStore.dispatch(action);
    const actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(1);
    expect(actions[0])
      .toEqual(action);
  });

  it('action with _REQUEST suffix will have _PENDING and _REQUEST dispatched', () => {
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    const actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(2);
    expect(actions)
      .toMatchSnapshot();
  });

  it(`handle ${_FULFILLED} correctly with payload`, () => {
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    let actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(2);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();

    const actionFullfill: ActionMeta<any, any> = {
      ...actions[0],
      type: `ACTION${_FULFILLED}`,
      payload: {
        data: 1,
        more_data: [1, 2, 3],
        even_more_data: {
          key: 'value',
        },
      },
    };
    mockStore.dispatch(actionFullfill);
    actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(3);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeTruthy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeTruthy();
    expect(actions[2].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeTruthy();
    expect(actions)
      .toMatchSnapshot();
  });

  it(`handle ${_REJECTED} correctly with payload`, () => {
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    let actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(2);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();

    const actionReject: ActionMeta<any, any> = {
      ...actions[0],
      type: `ACTION${_REJECTED}`,
      error: true,
      payload: new Error('Rejected!'),
    };
    mockStore.dispatch(actionReject);
    actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(3);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions[2].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions)
      .toMatchSnapshot();
  });

  it(`handle ${_ABORTED} correctly with payload`, () => {
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    let actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(2);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isFulfilled())
      .toBeFalsy();

    const actionReject: ActionMeta<any, any> = {
      ...actions[0],
      type: `ACTION${_ABORTED}`,
      error: true,
      payload: new Error('Aborted!'),
    };
    mockStore.dispatch(actionReject);
    actions = mockStore.getActions();
    (expect as any)(actions)
      .toHaveLength(3);
    expect(actions[0].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions[1].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions[2].meta[defaultOpts.metaKey].promise.isRejected())
      .toBeTruthy();
    expect(actions)
      .toMatchSnapshot();
  });
});

describe('Lets user set options', () => {
  it('metaKey', () => {
    const CUSTOM_META_KEY = 'Custom-Meta-Key';
    const mockStore = createMockStore({
      asyncFlowOpts: {
        metaKey: CUSTOM_META_KEY,
      },
    });
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    let actions = mockStore.getActions();
    const actionFullfill: ActionMeta<any, any> = {
      ...actions[0],
      type: `ACTION${_FULFILLED}`,
      payload: {
        data: 1,
        more_data: [1, 2, 3],
        even_more_data: {
          key: 'value',
        },
      },
    };
    mockStore.dispatch(actionFullfill);
    actions = mockStore.getActions();
    expect(actions)
      .toMatchSnapshot();
  });

  it('metaKeyRequestID', () => {
    const CUSTOM_META_KEY_REQUEST_ID = 'Custom-Meta-REQUEST-ID-Key';
    const mockStore = createMockStore({
      asyncFlowOpts: {
        metaKeyRequestID: CUSTOM_META_KEY_REQUEST_ID,
      },
    });
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    let actions = mockStore.getActions();
    const actionFullfill: ActionMeta<any, any> = {
      ...actions[0],
      type: `ACTION${_FULFILLED}`,
      payload: {
        data: 1,
        more_data: [1, 2, 3],
        even_more_data: {
          key: 'value',
        },
      },
    };
    mockStore.dispatch(actionFullfill);
    actions = mockStore.getActions();
    expect(actions)
      .toMatchSnapshot();
  });

  it('global request timeout', () => {
    const mockStore = createMockStore({
      asyncFlowOpts: {
        timeout: 5000,
      },
    });
    const action: Action<{}> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action);
    const actions = mockStore.getActions();
    expect(actions)
      .toMatchSnapshot();
  });

  it('per request timeout', () => {
    const mockStore = createMockStore();
    const action: ActionMeta<{}, any> = {
      type: `ACTION${_REQUEST}`,
      meta: {
        [defaultOpts.metaKey]: {
          timeoutRequest: 2500,
        } as IAsyncFlowActionMeta<any>,
      },
    };
    mockStore.dispatch(action);
    const actions = mockStore.getActions();
    expect(actions)
      .toMatchSnapshot();
  });

  it('generateId', () => {
    let idCounter = 2000;
    const mockStore = createMockStore({
      asyncFlowOpts: {
        generateId: ({ action }) => `${action.type}-${idCounter++}`,
      },
    });
    const action1: Action<any> = {
      type: `ACTION${_REQUEST}`,
    };
    const action2: Action<any> = {
      type: `ACTION${_REQUEST}`,
    };
    const action3: Action<any> = {
      type: `ACTION${_REQUEST}`,
    };
    mockStore.dispatch(action1);
    mockStore.dispatch(action2);
    mockStore.dispatch(action3);

    const actions = mockStore.getActions();
    expect(actions)
      .toMatchSnapshot();
  });
});
