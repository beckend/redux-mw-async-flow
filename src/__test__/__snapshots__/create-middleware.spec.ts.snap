exports[`Lets user set options generateId 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2000",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2000",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2001",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2001",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2002",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "ACTION_REQUEST-2002",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
]
`;

exports[`Lets user set options global request timeout 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "6-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 5000,
        "timeoutRequest": 5000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "6-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
]
`;

exports[`Lets user set options metaKey 1`] = `
Array [
  Object {
    "meta": Object {
      "Custom-Meta-Key": Object {
        "REQUEST_ID": "4-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "Custom-Meta-Key": Object {
        "REQUEST_ID": "4-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "meta": Object {
      "Custom-Meta-Key": Object {
        "REQUEST_ID": "4-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "payload": Object {
      "data": 1,
      "even_more_data": Object {
        "key": "value",
      },
      "more_data": Array [
        1,
        2,
        3,
      ],
    },
    "type": "ACTION_FULFILLED",
  },
]
`;

exports[`Lets user set options metaKeyRequestID 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "Custom-Meta-REQUEST-ID-Key": "5-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "Custom-Meta-REQUEST-ID-Key": "5-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "Custom-Meta-REQUEST-ID-Key": "5-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "payload": Object {
      "data": 1,
      "even_more_data": Object {
        "key": "value",
      },
      "more_data": Array [
        1,
        2,
        3,
      ],
    },
    "type": "ACTION_FULFILLED",
  },
]
`;

exports[`Lets user set options per request timeout 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "7-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 2500,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "7-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeoutRequest": 2500,
      },
    },
    "type": "ACTION_REQUEST",
  },
]
`;

exports[`Middleware action with _REQUEST suffix will have _PENDING and _REQUEST dispatched 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "0-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "0-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
]
`;

exports[`Middleware handle _ABORTED correctly with payload 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "3-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Aborted!],
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "3-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Aborted!],
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "error": true,
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "3-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Aborted!],
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "payload": [Error: Aborted!],
    "type": "ACTION_ABORTED",
  },
]
`;

exports[`Middleware handle _FULFILLED correctly with payload 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "1-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "1-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "1-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": Object {
            "data": 1,
            "even_more_data": Object {
              "key": "value",
            },
            "more_data": Array [
              1,
              2,
              3,
            ],
          },
          "isFulfilled": true,
          "isRejected": false,
          "rejectionReason": undefined,
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "payload": Object {
      "data": 1,
      "even_more_data": Object {
        "key": "value",
      },
      "more_data": Array [
        1,
        2,
        3,
      ],
    },
    "type": "ACTION_FULFILLED",
  },
]
`;

exports[`Middleware handle _REJECTED correctly with payload 1`] = `
Array [
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "2-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Rejected!],
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "type": "ACTION_PENDING",
  },
  Object {
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "2-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Rejected!],
        },
      },
    },
    "type": "ACTION_REQUEST",
  },
  Object {
    "error": true,
    "meta": Object {
      "asyncFlow": Object {
        "REQUEST_ID": "2-@@ASYNC_MIDDLEWARE-type--ACTION_REQUEST",
        "promise": Object {
          "fulfillmentValue": undefined,
          "isFulfilled": false,
          "isRejected": true,
          "rejectionReason": [Error: Rejected!],
        },
        "timeout": 10000,
        "timeoutRequest": 10000,
      },
    },
    "payload": [Error: Rejected!],
    "type": "ACTION_REJECTED",
  },
]
`;
